openshift/dynamic/client.py

```diff
@@ -738,7 +809,7 @@ def __iter__(self):
                             prefix, group, version, rg.preferred)
                         self._cache['resources'][prefix][group][version] = rg
                         self.__update_cache = True
-                    for resource in rg.resources:
+                    for _, resource in rg.resources.items():
```

Comment:
Good catch. Maybe use `iteritems` instead so you get an iterator rather than create a list.